cmd/yggd/client.go

```diff
@@ -185,3 +193,25 @@ func (c *Client) ConnectionStatus() (*yggdrasil.ConnectionStatus, error) {
 
 	return &msg, nil
 }
+
+func (c *Client) disconnectWorker(w worker) error {
```

Comment:
There's a separation of responsibilities conflict here. The `yggd.Client` is a relatively new data structure introduced into `yggd`, so I don't think its role has been clearly defined yet. However, I noticed that this PR pulled the `protocol` package into `client.go` as an import; this caught my attention, and it occurred to me that perhaps the `yggd.Client` should not be aware of the underlying RPC protocol used to communicate with workers. This is particularly important, considering the future plans to support multiple RPC protocols.

This method (`disconnectWorker`) looks like it could easily be moved into `grpc.go` as a method on `dispatcher`. That feels more like an appropriate place for this RPC-related method to exist. And then the code block within the `CommandNameDisconnect` case statement can call `c.d.disconnectWorker`.

Reply:
Good point; done.